### PR TITLE
[V0.4 Release] Docoment update for kubeflow and release notes

### PR DESCRIPTION
**Pull request by yds05**
1. Docoment update for kubeflow and release notes
2. update nni sdk's version from v0.3.0 to v0.4

### CODE REVIEW

File: docs/RELEASE.md [hunk 1]
```diff
@@ -1,3 +1,30 @@
...
+  * [Hyperband tuner](../src/sdk/pynni/nni/README.md#Hyperband)
+  * Support launch NNI experiment on MAC
+  * WebUI 
+    *  UI support for hyperband tuner
+    *  Remove tensorboard button 
+    *  Show experiment error message 
+    *  Show line numbers in search space and trial profile
+    *  Support search a specific trial by trial number
+    *  Show trial's hdfsLogPath
+    *  Download experiment parameters
```
Comment (chicm-ms) [inline docs/RELEASE.md]: "Download experiment results" would be better.

Comment (yds05) [reply docs/RELEASE.md]: Yes

File: docs/RELEASE.md [hunk 2]
```diff
@@ -1,3 +1,30 @@
...
+## Others
+  * Asynchronous dispatcher
+  * Docker file update, add pytorch library 
+  * Refactor 'nnictl stop' process, send SIGTERM to nni manager process, rather than calling stop Rest API. 
+  * OpenPAI training service bug fix
+    *  Support NNI Manager IP configuration(nniManagerIp) in PAI cluster config file, to fix the issue that user’s machine has no eth0 device 
+    *  File number in codeDir is capped to 1000 now, to avoid user mistakenly fill root dir for codeDir
+    *  Don’t print useless ‘metrics is empty’ log int PAI job’s stdout. Only print useful message once new metrics are recorded, to reduce confusion when user checks PAI trial’s output for debugging purpose
```
Comment (chicm-ms) [inline docs/RELEASE.md]: Don't print useless -> remove?

File: docs/RELEASE.md (hunk 1 above)
Comment (yds05) [reply docs/RELEASE.md]: Done

**Pull request merged**